functions/metadataStatusUpdater/src/test/dynamo.test.js

```diff
@@ -0,0 +1,86 @@
+const { expect } = require('chai');
+const sinon = require('sinon');
+const { DynamoDBDocumentClient, GetCommand, DeleteCommand, PutCommand } = require('@aws-sdk/lib-dynamodb');
+const { getItem, deleteItem, putMetadata } = require('../app/lib/dynamo');
+const { ItemNotFoundException } = require('../app/lib/exceptions');
+
+describe('dynamo.js tests', () => {
+
+  let docClientStub;
+  let logStub;
+
+  beforeEach(() => {
+    docClientStub = sinon.stub(DynamoDBDocumentClient.prototype, 'send');
+    logStub = sinon.stub(console, 'log');
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('getItem', () => {
+    it('should retrieve an item successfully', async () => {
+      const mockItem = { key: 'value' };
+      docClientStub.resolves({ Item: mockItem });
+
+      const result = await getItem('testTable', { id: 'testId' });
+
+      expect(docClientStub.firstCall.args[0]).to.be.an.instanceof(GetCommand);
+      expect(result).to.deep.equal(mockItem);
+    });
+
+    it('should throw ItemNotFoundException if item is not found', async () => {
+      docClientStub.resolves({ Item: undefined });
+
+      try {
+        await getItem('testTable', { id: 'testId' });
+      } catch (error) {
+        expect(error).to.be.an.instanceof(ItemNotFoundException);
+      }
+    });
+  });
+
+  describe('deleteItem', () => {
+    it('should delete an item successfully', async () => {
+      docClientStub.resolves({});
+
+      await deleteItem('testTable', { id: 'testId' });
+
+      expect(docClientStub.firstCall.args[0]).to.be.an.instanceof(DeleteCommand);
+    });
+  });
+
+  describe('putMetadata', () => {
```

Comment:
@empassaro qui mi sarei aspettato un test con un primo inserimento con un timestamp e un secondo inserimento con timestamp successivo per il caso ok e uno con un inserimento con timestamp precedente per il caso ko

Reply:
Sono d'accordo con l'idea alla base, però in questi test anche prendendo spunto da altre lambda che fanno uso di dynamo non mi sembra di aver visto questo approccio. Nel senso mi sembra che tutti abbiano mockato il client di dynamo, dunque non si è testato il "funzionamento" realistico come con i test con localstack.

Reply:
Ok allora durante i test e2e verifichiamo il comportamento corretto